<?xml version="1.0"?>
<launch>

	<param name="use_sim_time" value="true" />
                
	<node pkg="tf" type="static_transform_publisher" name="world_view" args="0 0 0 0 0 3.1415926 world world_view 100"/> 
	
	<node pkg="lidar_eskf" type="lidar_eskf_node" name="lidar_eskf" output="screen">
	
	    <remap from="/lidar_eskf/cloud"            to="/dji_sim/laser/pointcloud"/>
        
        <param name="map_file_name"            value="$(find lidar_eskf)/map/gazebo_bridge.bt"/>
        <param name="octree_resolution"        value="0.2"/>
        <param name="max_obstacle_dist"        value="3.0"/>
        <param name="ray_sigma"                value="1.0"/>
        <param name="cloud_resolution"         value="0.4"/>

	
	    <remap from="/imu"                      to="/dji_sim/imu"/>
	    <remap from="/meas"                     to="/lidar_eskf/posterior"/>
	    
        <param name="imu_frequency"             value="100"/>
        <param name="sigma_acceleration"        value="0.1"/>
        <param name="sigma_gyroscop"            value="0.01"/>
        <param name="sigma_acceleration_bias"   value="0.0002"/>
        <param name="sigma_gyroscope_bias"      value="0.0001"/>
        <param name="gravity"                   value="9.82"/>
        
        <param name="init_bias_acc_x"           value="0.0"/>
        <param name="init_bias_acc_y"           value="0.0"/>
        <param name="init_bias_acc_z"           value="0.0"/>
        
        <param name="acc_queue_size"            value="10"/>
        

	</node>
	
	<node pkg="octomap_server" type="octomap_server_node" name="octomap_server" 
	      args="$(find lidar_eskf)/map/gazebo_bridge.bt">
	      <param name="frame_id"           value="world"/>
	      <param name="base_frame_id"      value="world_view"/>
	</node>
	
	<node pkg="rviz" type="rviz" name="gpf_test_rviz" required="false" args="-d $(find lidar_eskf)/rviz/lidar_eskf.rviz"/>
</launch>
